addons/sale_project/models/project_project.py

```diff
@@ -45,6 +45,18 @@ def _domain_sale_line_id(self):
     reinvoiced_sale_order_id = fields.Many2one('sale.order', string='Sales Order', groups='sales_team.group_sale_salesman', domain="[('partner_id', '=', partner_id)]",
         help="Products added to stock pickings, whose operation type is configured to generate analytic costs, will be re-invoiced in this sales order if they are set up for it.",
     )
+    @api.model
+    def default_get(self, fields_list):
+        defaults = super().default_get(fields_list)
+        if self.env.context.get('state') == 'sale':
+            defaults['reinvoiced_sale_order_id'] = self.env.context.get('sale_project_sale_order_id')
+            sale_order_line_ids = self.env.context.get('sale_project_order_lines', [])
+            if sale_order_line_ids:
+                sale_order_lines = self.env['sale.order.line'].browse(sale_order_line_ids)
+                service_lines = sale_order_lines.filtered(lambda line: line.is_service)
+                if service_lines:
+                    defaults['sale_line_id'] = service_lines[0].id
+        return defaults
```

Comment:
```suggestion

@api.model
def default_get(self, fields_list):
    defaults = super().default_get(fields_list)
    if self._context.get('order_state') == 'sale':
        order_id = self._context.get('order_id')
        sale_line_id = self.env['sale.order.line'].search(
            [('order_id', '=', order_id), ('is_service', '=', True)], 
            order='id desc', limit=1
        ).id
        defaults.update({
	        'reinvoiced_sale_order_id': order_id,
	        'sale_line_id': sale_line_id,
	    })
    return defaults
```
I have not tested my suggestion.